### DISCUSSION TITLE
Connection made in submodule, searching times out, connection made in same file works.

### DISCUSSION THREAD

**Discussion by GerritKuilder** [Automattic/mongoose#14442] [Q&A]
Hi All.

I have the following working (nodejs)
a) index.js
connect to mongodb in the same index file, 
 with a function called connect() in index.js

const model = require ( ./models/model.js
Query can run and returns results.

b) in index.js, const connection=require( ./mongoconnect.js)
same connect() function as above.
Connection is made.(plenty of logging going on)
client=await connection.connect()
However the query now fails with a timeout.

Do I  need to export something extra or set a global variable?
```
async function connect(inConnection){
 client = mongoose.connect(inConnection, {
         maxPoolSize: 50,
            wtimeoutMS: 2500,
        });
        //await client.connect();
        console.log('Connected to the database');
[...]
}
```

Basically I am trying to create a npm module for local use so I can avoid duplicating code all over the place.
Any input appreciated.

Kind Regards,

Gerrit Kuilder

**Comment by GerritKuilder** [comment]
I might have cracked it:
Checked: https://github.com/Automattic/mongoose/blob/master/examples/express/connection-sharing/app.js

So in my main file I set the connection as  global.db= await mongo.connect(DATABASE_URL)

In my models I set my models as const track = global.db.model("track", trackSchema);

This works, have to be careful to create the connection first and then load the models.

Any improvements possible?

Kind Regards,

Gerrit Kuilder

**Reply by vkarpov15** [reply]
`global.db = await mongoose.connect()` works reasonably well, although we'd still recommend importing for readability purposes.

Do you have an example of the code structure from your original post? I'm not sure I follow exactly how your project is structured.
